src/modules/components/asset/assetTransfer/assetTransfer.tsx

```diff
@@ -0,0 +1,153 @@
+     */
+    hash: string;
+    /**
+     * Chain ID of the transaction.
+     */
+    chainId: number;
+}
+
+// could be moved to a utils file for export? reusable in TranactionDataListItem.Structure for example
+const formatValue = (tokenAmount: number, tokenSymbol: string, tokenPrice: number | string) => {
+    const formattedTokenValue = formatterUtils.formatNumber(tokenAmount, {
+        format: NumberFormat.TOKEN_AMOUNT_SHORT,
+        withSign: true,
+    });
+    const fiatValue = Number(tokenAmount) * Number(tokenPrice);
+    const formattedFiatValue = formatterUtils.formatNumber(fiatValue, { format: NumberFormat.FIAT_TOTAL_SHORT });
+    const formattedTokenAmount = `${formattedTokenValue} ${tokenSymbol}`;
+
+    return { formattedTokenAmount, formattedFiatValue };
+};
+
+export const AssetTransfer: React.FC<IAssetTransferProps> = (props) => {
+    const {
+        senderAddress,
+        recipientAddress,
+        senderEnsName,
+        recipientEnsName,
+        tokenName,
+        tokenIconSrc,
+        tokenAmount,
+        tokenSymbol,
+        tokenPrice,
+        chainId,
+        hash,
+        wagmiConfig: wagmiConfigProps,
+    } = props;
+    const wagmiConfigProvider = useConfig();
+
+    const wagmiConfig = wagmiConfigProps ?? wagmiConfigProvider;
+
+    const processedChainId = chainId ?? wagmiConfig.chains[0].id;
+
+    const currentChain = wagmiConfig.chains.find(({ id }) => id === processedChainId);
+    const blockExplorerUrl = currentChain?.blockExplorers?.default.url;
+
+    const blockExplorerAssembledHref = blockExplorerUrl && hash ? `${blockExplorerUrl}/tx/${hash}` : undefined;
+
+    const createLink = (address: Hash) => blockExplorerUrl && `${blockExplorerUrl}/address/${address}`;
+    const resolveHandle = (ensName: string | undefined, address: Hash) =>
+        ensName ?? addressUtils.truncateAddress(address);
+
+    const { formattedTokenAmount, formattedFiatValue } = formatValue(tokenAmount, tokenSymbol, tokenPrice);
+
+    const tokenTransferClassNames = classNames(
+        'flex h-16 w-full items-center justify-between rounded-xl border-[1px] border-neutral-100 px-4', // base
+        'hover:border-neutral-200 hover:shadow-neutral-md', // hover
+        'focus:outline-none focus-visible:rounded-xl focus-visible:ring focus-visible:ring-primary focus-visible:ring-offset', // focus
+        'active:border-neutral-300', // active
+        'md:h-20 md:px-6', // responsive
+    );
+
+    return (
+        <div className="flex h-full w-[320px] flex-col gap-y-2 md:w-[640px] md:gap-y-3">
+            <div className="relative flex h-full  flex-col rounded-xl border-[1px] border-neutral-100 md:flex-row">
+                <AssetTransferAddress
+                    txRole="sender"
+                    ensName={senderEnsName}
+                    address={senderAddress}
+                    link={createLink(senderAddress)}
+                    handle={resolveHandle(senderEnsName, senderAddress)}
```

Comment:
The logic related to the `AssetTransferAddress` component should be moved there, same for tests. I'd suggest to create a `/assetTransferAddress` subfolder and move the component and related tests there